src/providers/application/ApplicationProvider.tsx

```diff
@@ -239,24 +246,34 @@ function ApplicationProvider({ children }: ApplicationProviderProps) {
 
   async function submitApplication() {
     dispatch({ type: ApplicationActionType.LOADING });
-    const response = await fetch(
-      `/api/applications/${application!.id}/submit`,
-      {
-        method: "POST",
-      },
-    );
-    if (response.ok) {
-      fetchApplication();
-    } else {
-      const responseJson = await response.json();
-      const message = responseJson.detail || "Failed to submit application";
 
+    try {
+      const response = await fetch(
+        `/api/applications/${application!.id}/submit`,
+        { method: "POST" },
+      );
+
+      if (response.ok) {
+        dispatch({ type: ApplicationActionType.CLEAR_ERROR });
+        fetchApplication();
+      } else {
+        const errorResponse = await response.json();
+        const errorMessage =
+          errorResponse.error || "An unexpected error occurred.";
+        dispatch({
+          type: ApplicationActionType.REJECTED,
+          payload: errorMessage,
+        });
+      }
+    } catch (error) {
```

Comment:
here don't need catch because fetch dont throw excepton and we already check for error in else condition on line 259